pandas/core/arrays/string_arrow.py

```diff
@@ -237,7 +237,9 @@ def __init__(self, values):
             )
 
     @classmethod
-    def _from_sequence(cls, scalars, dtype: Dtype | None = None, copy: bool = False):
+    def _from_sequence(
+        cls, scalars, dtype: Dtype | None = None, copy: bool = False, coerce=True
```

Comment:
have we got tests for coerce=False. (to include MaskedArray)?

coerce=True, the default is identical to the old behaviour?

Reply:
coerce=True(old behavior) is tested elsewher(in astype, etc.), will add coerce=False tests.